iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1017,6 +1017,13 @@ private async void ShutdownAsync(IChannelHandlerContext context)
             }
             finally
             {
+                if (_passwordProvider is IotHubConnectionString iotHubConnectionString
+                    && iotHubConnectionString.TokenRefresher != null
+                    && iotHubConnectionString.TokenRefresher.InstanceCreatedBySdk)
+                {
+                    iotHubConnectionString.TokenRefresher.Dispose();
```

Comment:
if applicable, dispose the token refresher when the mqtt iot hub adapter shuts down.
Note that even though we have a token refresher associated with the mqtt connection, it isn't a "true" refresher in the sense that it doesn't monitor token expiration and doesn't initiate a preemptive renewal. The token here is refreshed only on CONNECT (if expired).